devtools/cli/src/main/java/io/quarkus/cli/Build.java

```diff
@@ -33,14 +33,23 @@ public Integer call() {
             output.throwIfUnmatchedArguments(spec.commandLine());
 
             BuildSystemRunner runner = getRunner();
+            if (buildOptions.generateReport) {
+                params.add("-Dquarkus.debug.dump-build-metrics=true");
+            }
             BuildSystemRunner.BuildCommandArgs commandArgs = runner.prepareBuild(buildOptions, runMode, params);
 
             if (runMode.isDryRun()) {
                 dryRunBuild(spec.commandLine().getHelp(), runner.getBuildTool(), commandArgs);
                 return CommandLine.ExitCode.OK;
             }
 
-            return runner.run(commandArgs);
+            int exitCode = runner.run(commandArgs);
+            if (exitCode == CommandLine.ExitCode.OK && buildOptions.generateReport) {
+                output.printText(new String[] {
+                        "\nBuild report generated: " + new BuildReport(runner).generate().toPath().toAbsolutePath().toString()
```

Comment:
Suggested this change mostly to single out the line as it is easy to miss.

```suggestion
                        "\nBuild report available: " + new BuildReport(runner).generate().toPath().toAbsolutePath().toString() + "\n"
```